compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -227,7 +227,12 @@ struct AbstractConstBuilder<'a, 'tcx> {
     tcx: TyCtxt<'tcx>,
     body: &'a mir::Body<'tcx>,
     /// The current WIP node tree.
-    nodes: IndexVec<NodeId, Node<'tcx>>,
+    ///
+    /// We require all nodes to be used in the final abstract const,
+    /// so we store this here. Note that we also consider nodes as used
+    /// if they are mentioned in an assert, so some used nodes are never
+    /// actually reachable by walking the [`AbstractConst`].
+    nodes: IndexVec<NodeId, (Node<'tcx>, bool)>,
```

Comment:
I think we need another table: `IndexVec<NodeId, Span>` in order to point to the actual problem instead of just the entire expression

Reply:
we could change this to `(Node<'tcx>, Span, bool)` or probably define a new struct for that